### ISSUE TITLE
Fail more gracefully when indexing into an empty frame with index. 

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#167]
When indexing into a frame with an index and no values, I would expect some form of KeyError or IndexError, as happens when the frame has data. Instead, the exception raised seems to change based on the type of the index:

```python
a = sf.Frame(index=list('abc'))
a.loc['a'] # builtins.AttributeError: 'numpy.str_' object has no attribute 'STATIC'

b = sf.Frame(index=range(3))
b.loc[0] # builtins.TypeError: object of type 'numpy.int64' has no len()
```
The same exceptions are raised when using iloc.

**Comment by ForeverWintr** [comment]
I suppose a second option is to return an empty Series for the calls above. This makes some sense if you think of a Frame with index but no columns as a collection of empty rows (and it's how Pandas handles the situation, for what that's worth).

**Comment by flexatone** [comment]
Many thanks for isolating this issue.

I think the best approach here is to return a zero-sized `Series`, as that is consistent with NumPy's handling of selection in similar cases:

```python
>>> np.array(()).reshape((0, 3))[:, 2]
array([], dtype=float64)
```

I have made the small change necessary to return the expected `Series` in these cases. This will be available in the 0.6.1 release.

```python
>>> a = sf.Frame(index=list('abc'))
>>> a.loc['a']
<Series: a>
<Index>
<int64>     <float64>
>>> b = sf.Frame(index=range(3))
>>> b.loc[0]
<Series>
<Index>
<int64>  <float64>
```
